provider/resource_entity_type.go

```diff
@@ -0,0 +1,682 @@
+		},
+	}
+}
+
+func getVitalMetricSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"metric_name": {
+			Type:     schema.TypeString,
+			Required: true,
+			Description: `The title of the vital metric. When creating vital metrics,
+					  it's a best practice to include the aggregation method and the name of the metric being calculated.
+					  For example, Average CPU usage.`,
+		},
+		"search": {
+			Type:     schema.TypeString,
+			Required: true,
+			Description: `The search that computes the vital metric. The search must specify the following fields:
+							- val for the value of the metric.
+							- _time because the UI attempts to render changes over time. You can achieve this by adding span={time} to your search.
+							- Fields as described in the split_by_fields configuration of this vital metric.
+							For example, your search should be split by host,region if the split_by_fields configuration is [ "host", "region" ].`,
+		},
+		"matching_entity_fields": {
+			Type:     schema.TypeMap,
+			Required: true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+			Description: `Specifies the aliases of an entity to use to match with the fields specified by the fields that the search configuration is split on.
+						Make sure the value matches the split by fields in the actual search.
+						For example:
+							- search = "..... by InstanceId, region"
+							- matching_entity_fields = {instance_id = "InstanceId", zone = "region"}.`,
+		},
+		"is_key": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  false,
+			Description: `Indicates if the vital metric specified is a key metric.
+						A key metric calculates the distribution of entities associated with the entity type to indicate the overall health of the entity type.
+						The key metric is rendered as a histogram in the Infrastructure Overview. Only one vital metric can have is_key set to true. `,
+		},
+		"unit": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "",
+			Description: "The unit of the vital metric. For example, KB/s. ",
+		},
+		"alert_rule": {
+			Type:     schema.TypeSet,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"suppress_time": {
+						Type:        schema.TypeString,
+						Optional:    true,
+						Default:     "0",
+						Description: "suppress the alert until this time",
+					},
+					"cron_schedule": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "frequency of alert search",
+					},
+					"is_enabled": {
+						Type:        schema.TypeBool,
+						Optional:    true,
+						Default:     false,
+						Description: "if alert is enabled",
+					},
+					"warning_threshold": {
+						Type:     schema.TypeInt,
+						Required: true,
+					},
+					"critical_threshold": {
+						Type:     schema.TypeInt,
```

Comment:
Can't this, as well as the `warning_threshold`, be a float number?

Reply:
Nope, at least UI argues with  'should be Int'

Reply:
```
Invalid literal for int() with base 10: '6.5'
```